src/Components/TaskScheduler.php

```diff
@@ -17,24 +19,95 @@ class TaskScheduler
     public static function schedule(SchedulerTask $task)
     {
         if (empty($task->payload)) {
-            $data = "";
+            $data = '';
         } else {
             $data = json_encode(json_decode($task->payload));
         }
+
         $verb = VerbsMask::toString($task->verb_mask);
         $serviceName = \ServiceManager::getServiceById($task->service_id)->getName();
         $component = $task->component;
         $commandOptions = '--format=JSON --verb=' . $verb . ' --service=' . $serviceName . ' --resource=' . $component;
 
         // Use the scheduler to schedule the task at its desired frequency in minutes
         if ($task->is_active) {
+
+            $dirPath = storage_path() . DIRECTORY_SEPARATOR . 'logs' . DIRECTORY_SEPARATOR . 'tmp';
+
+            if (!file_exists($dirPath)) mkdir($dirPath, 0777, true);
```

Comment:
I think it is a bad practice to create a directory with full access for all user

Reply:
I made it 0770. But this file will be deleted after file_get_contents function so I think it doesn't really matter.